src/schema/mod.rs

```diff
@@ -0,0 +1,339 @@
+    // Object validation keywords
+    //
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    properties: Vec<Property>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    keys: Option<Box<Schema>>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    values: Option<Box<Schema>>,
+    #[serde(default, rename = "additionalProperties")]
+    additional_properties: bool,
+    //
+    // StringList keywords
+    //
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    separator: Option<String>,
+    //
+    // Annotation keywords
+    //
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    title: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    help: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    warning: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    description: Option<String>,
+    //
+    // Array validation keywords
+    //
+    #[serde(
+        default,
+        skip_serializing_if = "Vec::is_empty",
+        deserialize_with = "deserialize_struct_or_vec"
+    )]
+    items: Vec<Schema>,
+    #[serde(default, rename = "maxItems", skip_serializing_if = "Option::is_none")]
+    max_items: Option<usize>,
+    #[serde(default, rename = "minItems", skip_serializing_if = "Option::is_none")]
+    min_items: Option<usize>,
+    #[serde(default, rename = "uniqueItems")]
+    unique_items: UniqueItems,
+    //
+    // Number validation keywords
+    //
+    #[serde(default, rename = "multipleOf", skip_serializing_if = "Option::is_none")]
+    multiple_of: Option<Number>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    max: Option<Number>,
+    #[serde(default, rename = "exclusiveMax", skip_serializing_if = "Option::is_none")]
+    exclusive_max: Option<Number>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    min: Option<Number>,
+    #[serde(default, rename = "exclusiveMin", skip_serializing_if = "Option::is_none")]
+    exclusive_min: Option<Number>,
+    //
+    // String validation keywords
+    //
+    #[serde(default, rename = "maxLength", skip_serializing_if = "Option::is_none")]
+    max_length: Option<usize>,
+    #[serde(default, rename = "minLength", skip_serializing_if = "Option::is_none")]
+    min_length: Option<usize>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_from_str"
+    )]
+    pattern: Option<Regex>,
+}
+
+impl Schema {
+    pub fn version(&self) -> u8 {
+        self.version.value()
+    }
+}
+
+//
+// Any instance type
+//
+impl Schema {
+    pub fn type_(&self) -> &Type {
+        &self.type_
+    }
+
+    pub fn const_(&self) -> Option<&Value> {
+        self.const_.as_ref()
+    }
+
+    pub fn default_(&self) -> Option<&Value> {
+        self.default_.as_ref()
+    }
+
+    pub fn enum_(&self) -> &[EnumEntry] {
```

Comment:
not sure what is he difference between `xxx_` and without `_` - is this because these collide with Rust keywords ? if so - would either having `_` everywhere help here , plus some comment ? Or potentially we could use the `r#` notation for those ? Like `r#enum` ?

Reply:
Oukey, replaced all `_` suffixes with `r#`. But man, it's much uglier :)